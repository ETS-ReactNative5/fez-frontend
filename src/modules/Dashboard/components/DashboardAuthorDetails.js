import React from 'react';
import PropTypes from 'prop-types';
import Avatar from 'material-ui/Avatar';

import DashboardResearcherIDs from './DashboardResearcherIDs';

const profileFallbackImage = require('../../../../public/images/avatar.svg');

const DashboardAuthorDetails = ({authorDetails}) => {
    return (
        <div className="columns userDetails">
            <div className="column is-narrow">
                <div className="accountHeadshot">
                    <Avatar size={150}
                            style={{
                                backgroundImage: `url("https://its-ss-uqresearchers.s3.amazonaws.com/photo/thumbnail_${authorDetails.uqr_id}.jpg"), url(${profileFallbackImage})`,
                                backgroundSize: 'cover, cover'
                            }}
                            backgroundColor="transparent"
                            aria-label={'Photograph of ' + authorDetails.title + ' ' + authorDetails.given_name + ' ' + authorDetails.family_name}
                            title={authorDetails.title + ' ' + authorDetails.given_name + ' ' + authorDetails.family_name}/>
                </div>
            </div>
            <div className="column is-narrow accountDetails">
                <div className="accountTitleName title is-3 color-reverse">
                    {authorDetails.title} {authorDetails.given_name} {authorDetails.family_name}
                </div>
                <div className="accountPositions column is-paddingless is-marginless is-narrow">

                    {authorDetails.positions.map((item, index) => (
                        <div key={index} className="accountPositionOrg color-reverse">
                            <strong>{item}</strong>
                            {authorDetails.org_units[index] ? ', ' : ''}
                            <span className="color-reverse">{authorDetails.org_units[index]}</span>
                        </div>
                    ))}

                </div>
                <DashboardResearcherIDs publonsId={authorDetails.publons_id}
                                        researcherId={authorDetails.researcher_id}
                                        scopusId={authorDetails.scopus_id}
                                        googleScholarId={authorDetails.google_scholar_id}
                                        orcidId={authorDetails.orcid_id}/>
            </div>
        </div>
    );
};

DashboardAuthorDetails.propTypes = {
    authorDetails: PropTypes.object.isRequired,
};

export default DashboardAuthorDetails;
