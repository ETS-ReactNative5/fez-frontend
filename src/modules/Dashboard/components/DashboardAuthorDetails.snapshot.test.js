jest.dontMock('./DashboardAuthorDetails');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import DashboardAuthorDetails from './DashboardAuthorDetails';

function setup({title, familyName, givenName, orgUnits, positions}) {
    const props = {
        title: 'title',
        familyName: 'family name',
        givenName: 'given name',
        orgUnits: ['1', '2', '3'],
        positions: ['1', '2', '3'],
    };
    return shallow(<DashboardAuthorDetails {...props} />);
}

describe('Alert snapshots test', () => {
    it('Render the logged in users details', () => {
        const title = 'title';
        const familyName = 'given name';
        const givenName = 'family name';
        const orgUnits = ['1', '2', '3'];
        const positions = ['1', '2', '3'];

        const wrapper = setup(title, familyName, givenName, orgUnits, positions);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
