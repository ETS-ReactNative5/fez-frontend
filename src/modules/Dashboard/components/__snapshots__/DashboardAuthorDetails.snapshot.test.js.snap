// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshots test Render the logged in users details 1`] = `
<div
  className="authorDetails"
>
  <div
    className="authorTitleName title is-3 color-reverse"
  >
    title
     
    given name
     
    family name
  </div>
  <div
    className="is-paddingless is-marginless is-narrow"
  >
    <div
      className="authorPositionOrg color-reverse"
    >
      <strong>
        1
      </strong>
      <span
        className="color-reverse"
      >
        , 
        1
      </span>
    </div>
    <div
      className="authorPositionOrg color-reverse"
    >
      <strong>
        2
      </strong>
      <span
        className="color-reverse"
      >
        , 
        2
      </span>
    </div>
    <div
      className="authorPositionOrg color-reverse"
    >
      <strong>
        3
      </strong>
      <span
        className="color-reverse"
      >
        , 
        3
      </span>
    </div>
  </div>
</div>
`;
